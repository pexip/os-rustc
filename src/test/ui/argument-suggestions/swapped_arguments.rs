struct X {}

fn two_args(_a: i32, _b: f32) {}
fn three_args(_a: i32, _b: f32, _c: &str) {}
fn four_args(_a: i32, _b: f32, _c: &str, _d: X) {}

fn main() {
  two_args(1.0, 1);
  //~^ ERROR mismatched types
  //~| ERROR mismatched types
  three_args(1.0,   1,  "");
  //~^ ERROR mismatched types
  //~| ERROR mismatched types
  three_args(  1,  "", 1.0);
  //~^ ERROR mismatched types
  //~| ERROR mismatched types
  three_args( "", 1.0,   1);
  //~^ ERROR mismatched types
  //~| ERROR mismatched types
  four_args(1.0, 1, X {}, "");
  //~^ ERROR mismatched types
  //~| ERROR mismatched types
  //~| ERROR mismatched types
  //~| ERROR mismatched types
}
