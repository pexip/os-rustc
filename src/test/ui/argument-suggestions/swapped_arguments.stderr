error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:8:12
   |
LL |   two_args(1.0, 1);
   |            ^^^ expected `i32`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:8:17
   |
LL |   two_args(1.0, 1);
   |                 ^
   |                 |
   |                 expected `f32`, found integer
   |                 help: use a float literal: `1.0`

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:11:14
   |
LL |   three_args(1.0,   1,  "");
   |              ^^^ expected `i32`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:11:21
   |
LL |   three_args(1.0,   1,  "");
   |                     ^
   |                     |
   |                     expected `f32`, found integer
   |                     help: use a float literal: `1.0`

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:14:20
   |
LL |   three_args(  1,  "", 1.0);
   |                    ^^ expected `f32`, found `&str`

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:14:24
   |
LL |   three_args(  1,  "", 1.0);
   |                        ^^^ expected `&str`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:17:15
   |
LL |   three_args( "", 1.0,   1);
   |               ^^ expected `i32`, found `&str`

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:17:26
   |
LL |   three_args( "", 1.0,   1);
   |                          ^ expected `&str`, found integer

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:20:13
   |
LL |   four_args(1.0, 1, X {}, "");
   |             ^^^ expected `i32`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:20:18
   |
LL |   four_args(1.0, 1, X {}, "");
   |                  ^
   |                  |
   |                  expected `f32`, found integer
   |                  help: use a float literal: `1.0`

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:20:21
   |
LL |   four_args(1.0, 1, X {}, "");
   |                     ^^^^ expected `&str`, found struct `X`

error[E0308]: mismatched types
  --> $DIR/swapped_arguments.rs:20:27
   |
LL |   four_args(1.0, 1, X {}, "");
   |                           ^^ expected struct `X`, found `&str`

error: aborting due to 12 previous errors

For more information about this error, try `rustc --explain E0308`.
